### PR TITLE
Weak Cartesian Morphisms

### PR DESCRIPTION
**Pull request by TOTBWF**
# Description

This PR defines weak cartesian morphisms, and proves some properties of them. It also proves some properties of cartesian morphisms, and defines classes of morphisms in displayed categories. It also pulls displayed isomorphisms into `Cat.Displayed.Morphism`.

### CODE REVIEW

File: src/Cat/Displayed/Cartesian.lagda.md [hunk 1]
```diff
@@ -129,6 +135,198 @@ composite, rather than displayed directly over a composite.
...
+    invr* = to-pathp⁻ $ cartesian→weak-monic f′-cart _ _ $
+      f′ ∘′ from* ∘′ to*      ≡⟨ shiftr (assoc _ _ _) (pulll′ _ (f′-cart .commutes _ _)) ⟩
+      hom[] (hom[] f″ ∘′ to*) ≡⟨ smashl _ _ ⟩
+      hom[] (f″ ∘′ to*)       ≡⟨ (hom[]⟩⟨ f″-cart .commutes _ _) ∙ hom[]-∙ _ _ ⟩
+      hom[] f′                ≡⟨ weave _ (sym $ idr _) (ap (_ ∘_) (sym $ idl _)) (symP $ idr′ f′) ⟩
+      hom[] (f′ ∘′ id′)       ≡˘⟨ whisker-r _ ⟩
+      f′ ∘′ hom[] id′ ∎
+```
+
+Cartesian morphisms are also stable under vertical vertical retracts.
```
Comment (plt-amy) [inline src/Cat/Displayed/Cartesian.lagda.md:262]: Doubly vertical?

File: src/Cat/Displayed/Cartesian/Weak.lagda.md [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+  field
+    {x′}    : Ob[ x ]
+    lifting : Hom[ f ] x′ y′
+    weak-cartesian : Weak-cartesian f lifting
+
+  open Weak-cartesian weak-cartesian public
+```
+
+A displayed category that has weak cartesian lifts for all morphisms
+in the base is called a "Prefibered Category". Notably, prefibered
```
Comment (plt-amy) [inline src/Cat/Displayed/Cartesian/Weak.lagda.md:182]: Use bold and just lowercase

File: src/Cat/Displayed/Cartesian/Weak.lagda.md [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+a cartesian lift of $f$.
+
+```agda
+  f-lift : Cartesian-lift f y′
+  f-lift .Cartesian-lift.x′ = x*
+  f-lift .Cartesian-lift.lifting = f*
+  f-lift .Cartesian-lift.cartesian = f*-cartesian
+```
+
+```agda
```
Comment (plt-amy) [inline src/Cat/Displayed/Cartesian/Weak.lagda.md:353]: Since there's no more prose after this can you wrap the rest of this module in an HTML comment?

**Review by plt-amy — CHANGES_REQUESTED** [review]

File: src/Cat/Displayed/Cartesian.lagda.md (hunk 1 above)
Comment (TOTBWF) [reply src/Cat/Displayed/Cartesian.lagda.md:262]: two for the price of one!

**Pull request merged**